src/main/java/de/photon/anticheataddition/modules/additions/esp/ThirdPersonCameraCanSee.java

```diff
@@ -1,64 +1,83 @@
-        final Vector lookingBehind = lookingForward.clone().multiply(-1);
-
-        // Do the Cameras intersect with Blocks?
-        // Get the length of the first intersection or 0 if there is none
-        double forwardDistance = getCameraDistance(eyeLocation, lookingForward);
-        double behindDistance = getCameraDistance(eyeLocation, lookingBehind);
+        // Calculate vectors for forward and backward cameras
+        final Vector forwardDir = direction.clone();
+        final Vector backwardDir = direction.clone().multiply(-1);
 
-        // All locations and camera rays go out from the eye location.
-        final Location[] result = new Location[]{eyeLocation, eyeLocation.clone(), eyeLocation.clone()};
+        // Calculate distances the cameras can travel before hitting blocks
+        final double forwardDistance = getCameraDistance(eyeLocation, forwardDir);
+        final double backwardDistance = getCameraDistance(eyeLocation, backwardDir);
 
-        // result[0] is the eye location and should not be changed.
-        // Subtract half a block to make sure that our intersection is before and not in a block.
-        result[1].add(lookingForward.normalize().multiply(forwardDistance));
-        result[2].add(lookingBehind.normalize().multiply(behindDistance));
+        // Create camera locations
+        final Location[] cameraLocations = new Location[3];
+        cameraLocations[0] = eyeLocation;
+        cameraLocations[1] = eyeLocation.clone().add(forwardDir.normalize().multiply(forwardDistance));
+        cameraLocations[2] = eyeLocation.clone().add(backwardDir.normalize().multiply(backwardDistance));
 
-        return result;
+        return cameraLocations;
     }
 
+    /**
+     * Determines whether the observing player can see the watched player
+     * from any third-person camera angle.
+     */
     @Override
-    public boolean canSeeTracing(Player observer, Player watched)
-    {
-        // ----------------------------------- Calculation ---------------------------------- //
-        final Location[] watchedHitboxLocations = Hitbox.espHitboxLocationOf(watched).getEspLocations();
+    public boolean canSeeTracing(Player observer, Player watched) {
+        final Location[] hitboxLocations = Hitbox.espHitboxLocationOf(watched).getEspLocations();
+        final Location[] cameraLocations = getCameraLocations(observer);
 
-        for (Location cameraLocation : getCameraLocations(observer)) {
-            final ResetVector between = new ResetVector(cameraLocation.toVector().multiply(-1));
-            for (Location hitLoc : watchedHitboxLocations) {
-                // Effectively hitLoc - cameraLocation because of the multiply(-1) above.
-                between.resetToBase().add(hitLoc.toVector());
+        // Check from each camera position
+        for (Location cameraLocation : cameraLocations) {
+            // Use negative camera position as base vector
```

Comment:
this is a great example for a bad code comment - it only explains what the code does, but not WHY.
what the code does is visible by looking at the code, so that does not need to be commented